### ISSUE TITLE
GQLDataFactory - Add GQL retry/resume/save/loading. Queries fail over time. 

### ISSUE THREAD

**Issue by idiom-bytes** [oceanprotocol/pdr-backend#436]
### Motivation
- When trying to get sufficient data from subgraph, errors happen.
- There are no retries/resuming, so you need to start over. Workflow to generate plots is poor.
- File deletion happens easily, improve this (especially important as we have multiple gql queries to make)
- Assume there are gql sources with different start dates and that choosing jan-01-2023 as st_ut doesn't break stuff

### Continue improving
- [x] Right now, do not delete file. Simply append-to-end only.

### DoD
- [x] Extend from cli-yaml2
- [x] Implement schema/validation so the data model is properly handled
- [x] Update timestamp logic such that ss/pp (in ms) are resolved against gql and data logic (i.e. timestamp filtering) whose timestamp may be in seconds, or even minutes.
- [x] Improve _update() such that file isn't deleted, and for now, will only be appended-to-end.
- [x] Implement logic + tests such that all subgraph-predictions data is managed by GQLDataFactory
- [x] Point existing stats/plot logic to use GQLDataFactory
- [x] Clean up DRY violations when trying to `get_sapphire_postfix()`
- [x] Created test/mock helper functions inside /pdr_backend/models/prediction.py
- [x] Created a new /pdr_backend/util/test_data.py that has all the common sample data that is shared by util data_eng tests
- [x] Reduced DRY violations around this as much as possible... however, data_eng/test/conftest.py and util/test/conftest.py still have duplicate fixtures as visible below... which I believe we could only address by not using fixtures...

### Do Next:
Please read [the following comments](https://github.com/oceanprotocol/pdr-backend/issues/436#issuecomment-1853104454) to organize next steps, and tackle the following items:

**Comment by idiom-bytes** [comment]
I think the last point Is "fairly done", without me doing more re-structuring....
- [ ]  Point existing stats/plot logic to use GQLDataFactory, remove code from utils, eliminate DRY for subgraph-predictions

Scripts we need to move to data_eng / analytics:
```
/util/get_predictions_info.py
/util/get_predictoors_info.py
/util/get_traction_info.py
/util/predictoor_stats.py
/util/subgraph_slot.py
/util/subgraph_predictions.py
```

As an outcome of this we should:
1. Move all aggregate/summary logic to a `/data_eng/data_pipeline/` folder
2. Move all plot logic to a `/analytics/plots/` folder
3. Move majority of gql_queries, over to data_factory
4. Update some of these queries such that the results could be cached/lookedup locally... such as with `get_all_contract_ids_by_owner` where 99% of cases we care about the latest OPF contracts
5. Rewrite all statistics and analytics to use the data lake and dataframes

**Comment by trentmc** [comment]
> I think the last point Is "fairly done", without me doing more re-structuring.... 

I just created a new epic [#447](https://github.com/oceanprotocol/pdr-backend/issues/447) and moved most of the checklist points into it. 

And in the top of the epic, I added another issue I recently created: "Rename/move files & dirs for proper separation among lake, AI models, analytics" [#446](https://github.com/oceanprotocol/pdr-backend/issues/446)

What I explicitly did not add: creating sub-directories of `lake/` or `analytics/`. This is premature optimization and adds unnecessary complexity. Shallow with more files in one directory is much easier to work with.
